server/komodo-server/src/main/java/org/komodo/rest/service/KomodoUtilService.java

```diff
@@ -968,108 +972,39 @@ public Response validateViewDefinition(final @Context HttpHeaders headers,
-            String validationVdbName = "validationVdbTemp";
-            String validationModelName = "views";
-            Model viewModel = null;
-            Vdb validationVdb = findVdb( uow, validationVdbName );
-            if(validationVdb == null) {
-                validationVdb = getWorkspaceManager(uow).createVdb( uow, null, validationVdbName, validationVdbName );
-            }
-            if (validationVdb.hasChild(uow, validationModelName, VdbLexicon.Vdb.DECLARATIVE_MODEL)) {
-            	validationVdb.removeModel(uow, validationModelName);
-            }
-            viewModel = validationVdb.addModel(uow, validationModelName);
-
-            // Run the DDL import
-            String msg = null;
-            ImportOptions importOptions = new ImportOptions();
-            ImportMessages importMessages = new ImportMessages();
-            try {
-				DdlImporter importer = new DdlImporter(repo);
-				importer.importDdl(uow, inputStream, viewModel, importOptions, importMessages);
-			} catch (Exception e) {
-				msg = e.getLocalizedMessage();
-			}
-            uow.commit();
-            
-            if (callback.await(3, TimeUnit.MINUTES)) {
-                if(msg == null) {
-                    List<String> errorMsgs = importMessages.getErrorMessages();
-                    if (errorMsgs.isEmpty()) {
-                        msg = "Validation successful";
-                    } else {
-                    	msg = errorMsgs.get(0);
-                    }
-                }
-            	viewDefnStatus.setStatus("SUCCESS");
-            	viewDefnStatus.setMessage(msg);
-            } else {
-            	viewDefnStatus.setStatus("ERROR");
-            	viewDefnStatus.setMessage("The validation timed out");
-            }
+        try {
+        	MetadataFactory mf = new MetadataFactory("foo", 1, "bar", SystemMetadata.getInstance().getRuntimeTypeMap(), new Properties(), null);
+        	QueryParser.getQueryParser().parseDDL(mf, restViewDefinition.getDdl());
+        	viewDefnStatus.setStatus("SUCCESS");
+        	viewDefnStatus.setMessage("View DDL was parsed successfully");
+		} catch (Exception ex) {
+			String msg = "Parsing Error for view: " + restViewDefinition.getViewName()
+				+ "\n" + ex.getMessage();
+        	LOGGER.error(msg);
+        	viewDefnStatus.setStatus("ERROR");
+        	viewDefnStatus.setMessage("Parsing Error\n" + ex.getMessage());
+		}
 
-        } catch ( final Exception e ) {
-            if ( ( uow != null ) && ( uow.getState() != State.COMMITTED ) ) {
-                uow.rollback();
-            }
-        	try {
-            	viewDefnStatus.setStatus("ERROR");
-            	viewDefnStatus.setMessage("Error with validation");
-			} catch (Exception e1) {
-			}
-        }
-        
-        cleanupValidationFile(inputStream, tmpFile);
+        UnitOfWork uow = null;
         
-        // create response
         try {
+        	SynchronousCallback callback = new SynchronousCallback();
```

Comment:
you dont need SynchronousCallback anymore - was using it before because there were 2 transactions.  Just do the commit like the other methods.  

return commit( uow, mediaTypes, viewDefnStatus );

also, rollbackOnly should be true on the createTransaction since you arent changing anything (think I had that wrong originally)

uow = createTransaction(principal, "Validate ViewDefinition", true );

Reply:
done